src/libvirtApi/domain.js

```diff
@@ -765,6 +765,44 @@ export function domainGetCapabilities({ connectionName, arch, model }) {
     return call(connectionName, '/org/libvirt/QEMU', 'org.libvirt.Connect', 'GetDomainCapabilities', ['', arch, model, '', 0], { timeout, type: 'ssssu' });
 }
 
+export function domainGetStartTime({
+    connectionName,
+    vmName,
+}) {
+    return cockpit.user().then(loggedUser => {
+        const options = { err: "message" };
+        if (connectionName === "system")
+            options.superuser = "try";
+
+        /* The possible paths of logfiles path, as of libvirt 9.3.0 are defined in:
+         * https://gitlab.com/libvirt/libvirt/-/blob/v9.3.0/src/qemu/qemu_conf.c#L153
+         * There libvirt defines 3 possible directories where they can be located
+         * - /root/log/qemu/ - That is however only relevant for "qemu:///embed", and not cockpit's use case
+         * - /var/log/libvirt/qemu/ - For privileged (qemu:///system) VMs
+         * - ~/.cache/libvirt/qemu/logs - For non-privileged  (qemu:///session) VMs
+         */
+        let logFile = `/var/log/libvirt/qemu/${vmName}.log`;
+        if (connectionName === "session")
+            logFile = `${loggedUser.home}/.cache/libvirt/qemu/log/${vmName}.log`;
+
+        // Use libvirt APIs for getting VM start up time when it's implemented:
+        // https://gitlab.com/libvirt/libvirt/-/issues/481
+        const args = ["grep", "'starting up'", `'${logFile}'`, "|", "tail", "-1"];
+
+        return cockpit.script(args.join(" "), options);
```

Comment:
This is a shell command, please just write it as a single string. Splitting it up into a vector is confusing, as `|` is not an argument of `grep`.